homeassistant/components/spotify/media_player.py

```diff
@@ -378,10 +378,13 @@ def update(self) -> None:
         current = self.data.client.current_playback()
         self._currently_playing = current or {}
 
-        self._playlist = None
         context = self._currently_playing.get("context")
-        if context is not None and context["type"] == MEDIA_TYPE_PLAYLIST:
-            self._playlist = self.data.client.playlist(current["context"]["uri"])
+        if context is not None and (
+            self._playlist is None or self._playlist["uri"] != context["uri"]
+        ):
+            self._playlist = None
+            if context["type"] == MEDIA_TYPE_PLAYLIST:
```

Comment:
Why do we not include the context type check in the check above before resetting the playlist?

Reply:
We will only get here when the stored playlist isn't matching what is played.
What is played now, doesn't have to have a playlist (but the existing/previous played song might have had one and needs to be cleared out).

Reply:
(its the else case)